packages/core/src/components/essentials/legend.ts

```diff
@@ -48,11 +72,24 @@ export class Legend extends Component {
 				return d.status === options.legend.items.status.ACTIVE;
 			});
 
-		addedLegendItems
-			.append("text")
-			.merge(legendItems.select("text"))
-			.html((d) => d.name)
-			.attr("alignment-baseline", "middle");
+		// truncate the legend label if it's too long
+		if (truncationTypeProvided !== TruncationTypes.NONE) {
```

Comment:
```js
addedLegendItems
	.append("text")
	.merge(legendItems.select("text"));
```

could move out of the if-statement and be stored in a variable

Reply:
Yes.